perma_web/api2/serializers.py

```diff
@@ -0,0 +1,242 @@
+        serializer = FolderSerializer(user.top_level_folders(), many=True)
+        return serializer.data
+
+
+### FOLDER ###
+
+class FolderSerializer(BaseSerializer):
+    has_children = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Folder
+        fields = ('id', 'name', 'parent', 'has_children', 'organization')
+        extra_kwargs = {'parent': {'required': True}}
+        allowed_update_fields = ['name', 'parent']
+
+    def get_has_children(self, folder):
+        return not folder.is_leaf_node()
+
+    def validate_name(self, name):
+        if self.instance:
+            # renaming
+            if self.instance.is_shared_folder:
+                raise serializers.ValidationError("Shared folders cannot be renamed.")
+            if self.instance.is_root_folder:
+                raise serializers.ValidationError("User's main folder cannot be renamed.")
+        return name
+
+    def validate_parent(self, parent):
+        if self.instance:
+            # moving
+            if self.instance.is_shared_folder:
+                raise serializers.ValidationError("Can't move organization's shared folder.")
+            if self.instance.is_root_folder:
+                raise serializers.ValidationError("Can't move user's main folder.")
+        return parent
+
+    def validate(self, data):
+        if 'name' in data:
+            # make sure folder name is unique in this location
+            parent_id = data['parent'].pk if 'parent' in data else self.instance.parent_id if self.instance else None
+            if parent_id:
+                unique_query = Folder.objects.filter(parent_id=parent_id, name=data['name'])
+                if self.instance:
+                    unique_query = unique_query.exclude(pk=self.instance.pk)
+                if unique_query.exists():
+                    raise serializers.ValidationError({'name':"A folder with that name already exists at that location."})
+        return data
+
+
+### ORGANIZATION ###
+
+class OrganizationSerializer(BaseSerializer):
+    registrar = serializers.StringRelatedField()
+    shared_folder = FolderSerializer()
+
+    class Meta:
+        model = Organization
+        fields = ('id', 'name', 'registrar', 'default_to_private', 'shared_folder')
+
+
+### CAPTUREJOB ###
+
+class CaptureJobSerializer(BaseSerializer):
+    guid = serializers.PrimaryKeyRelatedField(source='link', read_only=True)
+    class Meta:
+        model = CaptureJob
+        fields = ('guid', 'status', 'attempt', 'step_count', 'step_description', 'capture_start_time', 'capture_end_time', 'queue_position')
+
+
+### CAPTURE ###
+
+class CaptureSerializer(BaseSerializer):
+    class Meta:
+        model = Capture
+        fields = ('role', 'status', 'url', 'record_type', 'content_type', 'user_upload', 'playback_url')
+
+
+### LINK ###
+
+class LinkSerializer(BaseSerializer):
+    url = serializers.CharField(source='submitted_url', max_length=2100, required=False, allow_blank=True)
```

Comment:
It's too bad it doesn't inherit the max_length from the model by default. Oh well.